### DISCUSSION TITLE
Fail to configure Linux custom shortcuts to flameshot 10.2

### DISCUSSION THREAD

**Discussion by elad2109** [flameshot-org/flameshot#2163] [Q&A] [answered]
I downloaded an `AppImage` file for version 10.2 to `~/Applications/Flameshot` 
now I have to configure this cmd: `~/usr/local/bin/flameshot gui` so I can create a custom shortcut on Linux.
I'm not able to configure system custom shortcut for screenshoting via flameshot.

![image](https://user-images.githubusercontent.com/1093841/145274837-800521a1-66b5-437a-ab9e-159d04dc918d.png)

How can I fix this?

>> ~/usr/local/bin/flameshot gui

bash: /home/eladb/usr/local/bin/flameshot: No such file or directory

>> ~/Applications/Flameshot/flameshot gui

bash: /home/eladb/Applications/Flameshot/flameshot: No such file or directory
>> ls ~/Applications/Flameshot

Flameshot-0.10.2.x86_64.AppImage`

I must be missing something: 
$~/Downloads$ /usr/local/bin/flameshot gui  
bash: /usr/local/bin/flameshot: No such file or directory  

$:~/Downloads$ whereis flameshot  
flameshot:

**Comment by mmahmoudian** [comment]
You should use the full name:

```sh
${HOME}/Applications/Flameshot/Flameshot-0.10.2.x86_64.AppImage gui
```

If you want to avoid typing the full path in terminal, perhaps the best way is to create an alias. Open you `.bashrc` (or if you use ZSH open `.zshrc`) and add the following:

```sh
alias flameshot="${HOME}/Applications/Flameshot/Flameshot-0.10.2.x86_64.AppImage"
```

And then save the file, close your terminal and open again.

For creating a keyboard shortcut you should use the full path.

**Comment by elad2109** [comment]
Thanks

1) I've tried this, but it works only from the terminal and not when I set in in the custom shortcuts.

![image](https://user-images.githubusercontent.com/1093841/145904324-c6e8dacb-1e4a-4820-8876-d71e17933122.png)


2) Why this command worked for 10.1 installation on Debian?
`~/usr/local/bin/flameshot gui`

**Comment by mmahmoudian** [comment, marked as answer]
> but it works only from the terminal and not when I set in in the custom shortcuts.

I mentioned in the last line of my comment that "For creating a keyboard shortcut you should use the full path". What I meant is that you cannot use the `${HOME}` when creating the keyboard shortcut. I don't know your username so I had to use environmental variable. In terminal run `echo ${HOME}` to get the value of the `HOME` variable and replace the `${HOME}` in your shortcut with the value.

These are Linux knowledge and has nothing to do with Flameshot. Your might want to read a bit about variables in bash:

https://www.howtogeek.com/668503/how-to-set-environment-variables-in-bash-on-linux/

Note that to get the value of variables, conventionally you Would put the variable name in between `{}` (e.g `echo ${USER}`), but to get the value from commands you use `()` (e.g `echo $(date -I)`)

**Reply by elad2109** [reply]
Thanks! it worked.


Do you know why this command worked for 10.1 installation on Debian?
`~/usr/local/bin/flameshot gui`

**Reply by mmahmoudian** [reply]
Because you had the binary of Flameshot as `~/usr/local/bin/flameshot` or you had a symlink file to the AppImage. If you do `ls -al ~/usr/local/bin/flameshot` you can see if the file exists and if it is a symlink file. If it is a symlink, the file permission will start with L as shown in this picture:

![image](https://user-images.githubusercontent.com/390889/146008674-c3c3d853-bcef-497b-b5ee-bb7d097f97f0.png)
